drivers/soundwire/bus.c

```diff
@@ -1082,12 +1082,125 @@ int sdw_configure_dpn_intr(struct sdw_slave *slave,
 	return ret;
 }
 
+static int sdw_slave_set_frequency(struct sdw_slave *slave)
+{
+	u32 mclk_freq = slave->bus->prop.mclk_freq;
+	u32 curr_freq = slave->bus->params.curr_dr_freq >> 1;
+	u8 base = 0;
+	u8 scale_index;
+	u32 scale;
```

Comment:
I'd say `scale` should be just an `unsigned int`

Reply:
yes, fixed

Reply:
both fixed